code/datums/components/bullet_intercepting.dm

```diff
@@ -0,0 +1,67 @@
+/**
+ * Component which allows an equipped item to occasionally absorb a projectile.
+ */
+/datum/component/bullet_intercepting
```

Comment:
It seems as if you've recreated `/obj/item/proc/hit_reaction`

Reply:
hit_reaction gets used for way more than projectiles

Reply:
yes but you can narrow down hit reaction to only respond to projectiles (`if(attack_type != PROJECTILE_ATTACK) return`

Reply:
It looks like `hit_reaction` is only called for held items, suits, shirts, neck items, and hats.
I could just make it loop every item if you'd prefer that to the component, but then I'm going to be copy/pasting more code.